### PR TITLE
Stop skipping advanced clone method tests with immediate bind annotation

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
- Smart/CSI clone
These tests are by definition only running on CSI storage (IsCSIVolumeCloneStorageClassAvailable/IsSnapshotStorageClassAvailable)
- Snapshot clone
This test is always getting skipped, since snapshots always imply the storage is CSI

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/cloner_test.go [hunk 1]
```diff
@@ -2822,13 +2805,8 @@ var _ = Describe("all clone tests", func() {
 
 				By("Waiting for clone to be completed")
 				Expect(utils.WaitForDataVolumePhase(f, dataVolume.Namespace, cdiv1.Succeeded, dataVolume.Name)).To(Succeed())
-				By("Check host assisted clone is taking place")
 				pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				suffix := "-host-assisted-source-pvc"
-				Expect(pvc.Annotations[controller.AnnCloneRequest]).To(HaveSuffix(suffix))
-				Expect(pvc.Spec.DataSource).To(BeNil())
-				Expect(pvc.Spec.DataSourceRef).To(BeNil())
```
Comment (arnongilboa) [inline tests/cloner_test.go:2831]: why was it removed?

Comment (akalenyu) [reply tests/cloner_test.go:2831]: we no longer fall back to host assisted - https://github.com/kubevirt/containerized-data-importer/pull/2765
so now this test is just about the result being a success

**Pull request merged**